src/typing/env_js.ml

```diff
@@ -29,6 +29,10 @@ let refinement_key names =
 let is_refinement name =
   (String.length name) >= 5 && (String.sub name 0 5) = "$REFI"
 
+let refined_object_from_key name =
+  let i = String.index name '.' in
+  String.sub name 6 (i - 6)
```

Comment:
This is only used in `install_refinement`, which is called from `Env_js` where we've already verified `is_refinement name`. I agree this is ugly, though. I would like to encapsulate this more, but I feel like adding a dependency from `Env_js` to `Type_inference_js` is in the wrong direction here.